pkg/controller/kinesis/stream/setup.go

```diff
@@ -124,7 +124,7 @@ func postCreate(_ context.Context, cr *svcapitypes.Stream, obj *svcsdk.CreateStr
 	if err != nil {
 		return managed.ExternalCreation{}, err
 	}
-	meta.SetExternalName(cr, cr.Name)
+	meta.SetExternalName(cr, *awsclients.String(meta.GetExternalName(cr)))
```

Comment:
This would have no effect and just writes the value of the field into itself.

Given that in `preCreate` it is assumed that the external name is already set (and where it is actually needed), there is no need to set in `postCreate`.

I agree that using `cr.Name` makes no sense and is against the convention (that external name should be used). If we want to use the external name as default if external name is not set, we should use the `managed.NewNameAsExternalName` initializer as it is done in other places like https://github.com/crossplane-contrib/provider-aws/blob/master/pkg/controller/batch/computeenvironment/setup.go#L61.